tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAsset.scala

```diff
@@ -0,0 +1,76 @@
+package com.linagora.tmail.james.jmap.publicAsset
+
+import java.net.URI
+import java.util.UUID
+
+import com.linagora.tmail.james.jmap.publicAsset.ImageContentType.ImageContentType
+import eu.timepit.refined
+import eu.timepit.refined.api.{Refined, Validate}
+import org.apache.james.jmap.api.model.IdentityId
+import org.apache.james.jmap.api.model.Size.Size
+import org.apache.james.jmap.core.AccountId
+import org.apache.james.jmap.mail.BlobId
+import org.apache.james.mailbox.model.ContentType
+
+object PublicAssetIdFactory {
+  def generate(): PublicAssetId = PublicAssetId(UUID.randomUUID())
+}
+
+case class PublicAssetId(value: UUID) {
+  def asString(): String = value.toString
+}
+
+object PublicURI {
+  def fromString(value: String): PublicURI = PublicURI(new URI(value))
+
+  def from(assetId: PublicAssetId, account: AccountId, publicURIPrefix: String = ""): PublicURI =
```

Comment:
I would require `publicURIPrefix` to be a URI and use URI methods rahter than string interpolation...